boards/nxp/imx93_evk/board.c

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2024 NXP
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/gpio.h>
+
+LOG_MODULE_REGISTER(board_control, CONFIG_BOARD_MIMX93_EVK_A55_LOG_LEVEL);
+
+#if !defined(CONFIG_BOARD_MIMX93_EVK_A55_EXP_SEL_NO_INIT)
+
+#define MIMX93_EVK_A55_EXP_SEL_DEV DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio_exp0))
+#define MIMX93_EVK_A55_EXP_SEL_PIN (4U)
+
+static int board_init_exp_sel(void)
+{
+	int rc = 0;
+	const struct device *exp_dev = MIMX93_EVK_A55_EXP_SEL_DEV;
+	const uint32_t exp_sel_pin = MIMX93_EVK_A55_EXP_SEL_PIN;
+	uint32_t pin_state;
+
+	LOG_INF("i.mx93 board mux init");
+	if (exp_dev == NULL) {
+		LOG_ERR("adp5585 expander not found, aborting");
+		return -ENODEV;
+	} else if (!device_is_ready(exp_dev)) {
+		LOG_ERR("%s is not ready", exp_dev->name);
+		return -ENODEV;
+	}
+	rc = gpio_pin_configure(exp_dev, exp_sel_pin, GPIO_OUTPUT);
+	if (rc) {
+		LOG_ERR("error %d while configuring pin", rc);
+		return rc;
+	}
+
+#if defined(CONFIG_BOARD_MIMX93_EVK_A55_EXP_SEL_MUX_A)
+	pin_state = 0U;
+#elif defined(CONFIG_BOARD_MIMX93_EVK_A55_EXP_SEL_MUX_B)
+	pin_state = 1U;
+#endif
+	LOG_INF("exp_sel mux %c with priority %d",
+		pin_state ? 'B' : 'A',
+		CONFIG_BOARD_MIMX93_EVK_A55_EXP_SEL_INIT_PRIO
+	);
+
+	rc = gpio_pin_set_raw(exp_dev, exp_sel_pin, pin_state);
+	if (rc) {
+		LOG_ERR("error %d while set pin to %d", rc, pin_state);
+		return rc;
+	}
+
+	return 0;
+}
+
+SYS_INIT(board_init_exp_sel, POST_KERNEL, CONFIG_BOARD_MIMX93_EVK_A55_EXP_SEL_INIT_PRIO);
```

Comment:
We've tried that. gpio-hog will require adding overlay to each samples / tests, which is not scalable. So I come up with this idea to control gpio with Kconfig.

Reply:
> My opinion is, given the current implementation of gpio_hog in zephyr, it can barely handle any init dependancies. 

Right. The gpio-hog mechanism is for static GPIO configuration at boot. It's not useful if you need to be able to change e.g. a GPIO-controlled MUX between multiple configurations. 

I'd suggest adding a local, board-specific MUX devicetree binding and control this from `board.c` based on _existing_ Kconfig symbols.